tests/cpp/test_multidevice_communications.cpp

```diff
@@ -414,9 +414,13 @@ INSTANTIATE_TEST_SUITE_P(
     testing::Values(CommunicatorBackend::kNccl),
     testing::PrintToStringParamName());
 
-using P2PCommunicationTest = MultiDeviceTest;
+using P2PCommunicationTestParams = std::string;
 
-TEST_F(P2PCommunicationTest, CudaComm) {
+class P2PCommunicationTest
+    : public MultiDeviceTest,
+      public testing::WithParamInterface<P2PCommunicationTestParams> {};
```

Comment:
```suggestion
      public testing::WithParamInterface<P2pProtocol> {};
```
which is more type-safe than a string. PrintToStringParamName should work too since you have defined operator<<.